Test/NativeUnitTest/NativeUnitTest.cpp

```diff
@@ -1,23 +1,88 @@
 #include "pch.h"
 #include "CppUnitTest.h"
+#include "../Test/ConsoleReader.cpp"
 #include "../Test/Source.cpp"
+#include <isolator.h>
 
 using namespace Microsoft::VisualStudio::CppUnitTestFramework;
 
 namespace NativeUnitTestForMain
 {
 	TEST_CLASS(MainTests)
 	{
+		std::string path = "D:/bruhmoment/Abyss/test.txt";
+		int charOccurs[possibleChars] = {};
 	public:
 		
-		TEST_METHOD(TestIfPathExists)
+		TEST_METHOD(TestIfInputPathExists)
 		{
-			int charOccurs[possibleChars];
-			for (int i = 0; i < possibleChars; ++i) {
-				charOccurs[i] = 0;
-			}
-			Assert::IsFalse(calculateOccursInFile("randompath.txt", {}));
-			Assert::IsTrue(calculateOccursInFile("D:/bruhmoment/Abyss/ToDo.txt", charOccurs));
+			Assert::IsTrue(calculateOccursInFile(path, charOccurs));
+		}
+		TEST_METHOD(TestIfOutputPathExists)
+		{
+			Assert::IsTrue(calculateOccursInFile(path, charOccurs));
+		}
+		TEST_METHOD(TestCalculateOccursInLine) 
+		{
+			std::string line = "aaA  ";
+			calculateOccurs(charOccurs, line);
+			Assert::AreEqual(2, charOccurs[int('a')]);
+		}
+		TEST_METHOD(TestConsoleReader)
+		{
+			// ConsoleReader* reader = FAKE<ConsoleReader>();
+			ConsoleReader reader;
+			FAKE_GLOBAL(_getch);
+			char testChars[] = {'a', 'b', backspace, 'c', enter};
+
+
+			// int * timesCalled = TIMES_CALLED(_getch());
+			// WHEN_CALLED(_getch()).Return(testChars[int(timesCalled)]);
+			// why doesn't this work ;(
+
+
+			WHEN_CALLED(_getch()).Return(testChars[0]);
+			WHEN_CALLED(_getch()).Return(testChars[1]);
+			WHEN_CALLED(_getch()).Return(testChars[2]);
+			WHEN_CALLED(_getch()).Return(testChars[3]);
+			WHEN_CALLED(_getch()).Return(testChars[4]);
+			reader.readConsole("testing some letters with backspace and enter");
+			Assert::AreEqual(2, int(reader.getData().length()));
+
+
+			WHEN_CALLED(_getch()).Return(esc);
+			reader.readConsole("testing escape");
+			Assert::AreEqual(0, int(reader.getData().length()));
+		}
+		 TEST_METHOD(TestCalculateOccursInFile)
+		{
+			// you can't fake ifstream with this framework :/
+			 std::ofstream testFile;
+			 std::string testPath = "testFile.txt";
+			 std::string testData = "testData";
+			 testFile.open(testPath);
+			 testFile << testData;
+			 Assert::IsTrue(testFile.is_open());
+			 testFile.close();
+
+
+			 int manualCharOccurs[possibleChars] = {};
+			 int testCharOccurs[possibleChars] = {};
+			 calculateOccurs(manualCharOccurs, testData);
+			 Assert::IsTrue(calculateOccursInFile(testPath, testCharOccurs));
+
+
+			 for (int i = 0; i < possibleChars; ++i) {
+				 Assert::AreEqual(manualCharOccurs[i], testCharOccurs[i]);
+			 }
+			 // comparing arrays doesn't work, have to compare each element
```

Comment:
It works :)
If you tried to compare like manualCharOccurs == testCharOccurs, then you are wrong and don't know how to compare arrays. Explain me what do you actually compare with such check?
At least you can compare to memory blocks with memcp

Reply:
> Explain me what do you actually compare with such check
You compare memory addresses of the beginnings of arrays (I think?)
I just didn't want to include any additional libraries, so the lööp seemed like the best option.

But I guess std::equal() exists, so I'll use that